### PR TITLE
lint: enable more pydocstyle rules

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Enables all relevant `D4*` lint rules, ignoring the ones that aren't usable or only apply to google-style docstrings.
Little to no changes in resulting documentation.

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `pdm lint`
    - [ ] I have type-checked the code by running `pdm pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/ext/commands/base_core.py [hunk 1]
```diff
@@ -380,8 +380,8 @@ def get_cooldown_retry_after(self, inter: ApplicationCommandInteraction) -> floa
 
         return 0.0
 
+    # This method isn't really usable in this class, but it's usable in subclasses.
     async def invoke(self, inter: ApplicationCommandInteraction, *args, **kwargs) -> None:
-        """This method isn't really usable in this class, but it's usable in subclasses."""
```
Comment (onerandomusername) [inline disnake/ext/commands/base_core.py:384]: this should continue to be documented

Comment (shiftinv) [reply disnake/ext/commands/base_core.py:384]: `Command.invoke` and `Group.invoke` are also not documented, all of these should be implementation details.

Comment (onerandomusername) [reply disnake/ext/commands/base_core.py:384]: sounds fair.

Comment (onerandomusername) [reply disnake/ext/commands/base_core.py:384]: we should mention something about this method now being undocumented though.

Comment (shiftinv) [reply disnake/ext/commands/base_core.py:384]: https://github.com/DisnakeDev/disnake/pull/961/commits/87d27a96e5ead52c350b8944b76d88a1f996808c

**Review by onerandomusername — APPROVED** [review]

**Pull request merged**